cts-java/src/test/java/org/ga4gh/cts/api/variants/CallSetsPagingIT.java

```diff
@@ -0,0 +1,150 @@
+
+        final String variantSetId = Utils.getVariantSetId(client);
+        // retrieve them all - this may return fewer than "all," however.
+        final List<CallSet> listOfCallSets = Utils.getAllCallSets(client, variantSetId);
+        assertThat(listOfCallSets).isNotEmpty();
+
+        // we will do a set comparison after retrieving them 1 at a time
+        final Set<CallSet> setOfExpectedCallSets = new HashSet<>(listOfCallSets);
+        assertThat(listOfCallSets).hasSameSizeAs(setOfExpectedCallSets);
+
+        final Set<CallSet> setOfCallSetsGathered1By1 = new HashSet<>(setOfExpectedCallSets.size());
+
+        // page through the ReferenceSets using the same query parameters and collect them
+        String pageToken = null;
+        do {
+            final SearchCallSetsRequest pageReq =
+                    SearchCallSetsRequest.newBuilder()
+                                         .setVariantSetId(variantSetId)
+                                         .setPageSize(1)
+                                         .setPageToken(pageToken)
+                                         .build();
+            final SearchCallSetsResponse pageResp =
+                    client.variants.searchCallSets(pageReq);
+            final List<CallSet> pageOfCallSets = pageResp.getCallSets();
+            pageToken = pageResp.getNextPageToken();
+
+            assertThat(pageOfCallSets).hasSize(1);
+            setOfCallSetsGathered1By1.add(pageOfCallSets.get(0));
+
+        } while (pageToken != null);
+
+        assertThat(setOfCallSetsGathered1By1).containsAll(setOfExpectedCallSets);
+    }
+
+    /**
+     * Check that we can page through the {@link CallSet}s we receive from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.Variants#searchCallSets(SearchCallSetsRequest)}
+     * using an increment as large as the non-paged set of results.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingByOneChunkThroughCallSets() throws AvroRemoteException {
+        final String variantSetId = Utils.getVariantSetId(client);
+        final List<CallSet> listOfCallSets = Utils.getAllCallSets(client, variantSetId);
+
+        // page through the CallSets in one gulp
+        checkSinglePageOfCallSets(variantSetId,
+                                  listOfCallSets.size(),
+                                  listOfCallSets);
+    }
+
+    /**
+     * Check that we can page through the CallSets we receive from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.Variants#searchCallSets
+     * (SearchCallSetsRequest)}
+     * using an increment twice as large as the non-paged set of results.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link
+     * GAException})
+     */
+    @Test
+    public void checkPagingByOneTooLargeChunkThroughCallSets() throws AvroRemoteException {
```

Comment:
This test implicitly asserts that default page returned by the server includes all results. As discussed previously (and as expressed in the comment @ line 51) this is not necessarily true.